src/components/app/data/hooks/useCourseMetadata.test.jsx

```diff
@@ -138,41 +183,54 @@ describe('useCourseMetadata', () => {
     useLateEnrollmentBufferDays.mockReturnValue(undefined);
     useSearchParams.mockReturnValue([new URLSearchParams({})]);
 
-    const mockCourseRuns = [
-      ...mockCourseMetadata.courseRuns,
-      {
-        ...mockCourseMetadata.courseRuns[0],
-        key: 'course-v1:edX+DemoX+2018',
+    const availableCourseRuns = [
+      mockCourseMetadata.courseRuns[0], // This is marketable, enrollable, unrestricted, etc.
+      mockCourseMetadata.courseRuns[2], // This is restricted, but that doesn't disqualify it.
+    ];
+    const unavailableCourseRuns = [
+      mockCourseMetadata.courseRuns[1], // This one is unavailable due to being unmarketable.
+    ];
+    // Copy all the above runs to make both assigned and unassigned versions.
+    const courseRunsMatrix = {
+      available: {
+        assigned: availableCourseRuns.map(r => ({ ...r, key: `${r.key}assigned` })),
+        unassigned: availableCourseRuns.map(r => ({ ...r, key: `${r.key}unassigned` })),
       },
+      unavailable: {
+        assigned: unavailableCourseRuns.map(r => ({ ...r, key: `${r.key}assigned` })),
+        unassigned: unavailableCourseRuns.map(r => ({ ...r, key: `${r.key}unassigned` })),
+      },
+    };
```

Comment:
[nit/curious] I think this is a great idea to defined a set of course runs which are considered assigned and unassigned, but I am wondering if we should be dependent on the mocked value for redeemablePolicies to determine if a specific run is assigned or unassigned.

Reply:
If I'm understanding correctly, you're suggesting that the mocked value for useRedeemablePolicies should match the intent of this matrix? Below on line 234 I'm overriding the useRedeemablePolicies mock to do basically that.